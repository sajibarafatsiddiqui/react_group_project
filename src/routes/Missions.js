import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchAllMissions } from 'redux/missions/missionsSlice';
import MissionList from '../components/MissonList';

const Missions = () => {
  const dispatch = useDispatch();
  const { missions } = useSelector((store) => store.missions);
  useEffect(() => {
    dispatch(fetchAllMissions());
  }, [dispatch]);
  return (
    <>
      <MissionList missions={missions} />
    </>
  );
};

export default Missions;
