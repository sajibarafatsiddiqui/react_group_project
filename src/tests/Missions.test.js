import { configureStore, createSlice } from '@reduxjs/toolkit';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Missions from '../routes/Missions';

describe('Tests for <MissionList /> component', () => {
  const initialState = {
    missions: [{
      missionId: '1',
      missionName: 'Mission 1',
      description: 'Mission Desc 1',
      reserved: false,
    }, {
      missionId: '2',
      missionName: 'Mission 2',
      description: 'Mission Desc 2',
      reserved: true,
    }],
    isLoading: false,
  };

  const missionSlice = createSlice({
    name: 'missions',
    initialState,
  });

  const store = configureStore({
    reducer: {
      missions: missionSlice.reducer,
    },
  });

  const wrapper = (component) => (
    <Provider store={store}>
      <>
        {component}
      </>
    </Provider>
  );

  it('Should render two missions', () => {
    render(wrapper(<Missions />));
    expect(screen.queryByTestId('missions-container').children.length).toBe(2);
  });

  it('Should match the snapshot', () => {
    expect(renderer.create(wrapper(<Missions />)).toJSON()).toMatchSnapshot();
  });
});
