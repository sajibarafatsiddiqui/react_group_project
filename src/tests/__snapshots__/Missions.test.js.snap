// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <MissionList /> component Should match the snapshot 1`] = `
<div
  className="p-4"
>
  <table
    className="table-responsive table table-sm table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th
          className="col-md-1"
        >
          Mission
        </th>
        <th
          className=""
        >
          Description
        </th>
        <th
          className="col-sm-1"
        >
          Status
        </th>
        <th
          className="col-md-1"
        >
          Â 
        </th>
      </tr>
    </thead>
    <tbody
      data-testid="missions-container"
    >
      <tr>
        <td>
          <strong>
            Mission 1
          </strong>
        </td>
        <td>
          Mission Desc 1
        </td>
        <td
          className="align-middle"
        >
          <button
            className="btn btn-secondary btn-sm"
            disabled={false}
            type="button"
          >
            NOT A MEMBER
          </button>
        </td>
        <td
          className="align-middle"
        >
          <button
            className="btn btn-outline-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <strong>
            Mission 2
          </strong>
        </td>
        <td>
          Mission Desc 2
        </td>
        <td
          className="align-middle"
        >
          <button
            className="btn btn-success btn-sm"
            disabled={false}
            type="button"
          >
            Active Member
          </button>
        </td>
        <td
          className="align-middle"
        >
          <button
            className="btn btn-outline-danger"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Leave Mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
